crates/ark/src/lsp/completions/sources/composite/call.rs

```diff
@@ -37,54 +36,34 @@ impl CompletionSource for CallSource {
         &self,
         completion_context: &CompletionContext,
     ) -> anyhow::Result<Option<Vec<CompletionItem>>> {
-        completions_from_call(
-            completion_context.document_context,
-            completion_context.pipe_root(),
-        )
+        completions_from_call(completion_context)
     }
 }
 
 fn completions_from_call(
-    context: &DocumentContext,
-    root: Option<PipeRoot>,
+    context: &CompletionContext,
 ) -> anyhow::Result<Option<Vec<CompletionItem>>> {
-    let mut node = context.node;
-    let mut has_call = false;
-
-    loop {
-        // If we landed on a 'call', then we should provide parameter completions
-        // for the associated callee if possible.
-        if node.is_call() {
-            has_call = true;
-            break;
-        }
-
-        // If we reach a brace list, bail.
-        if node.is_braced_expression() {
-            break;
-        }
-
-        // Update the node.
-        node = match node.parent() {
-            Some(node) => node,
-            None => break,
-        };
-    }
-
-    if !has_call {
-        // Didn't detect anything worth completing in this context,
-        // let other sources add their own candidates instead
+    if !context.is_in_call() {
+        // Not in a call, let other sources add their own candidates
         return Ok(None);
     }
 
+    let document_context = context.document_context;
+    let point = document_context.point;
+
+    // Find the call node
```

Comment:
I think I'm actually going to store this as a field of `CompletionContext`. Coming soon.

Reply:
Done!